engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceMigrationMigrateProcessor.java

```diff
@@ -177,6 +188,16 @@ private void tryMigrateElementInstance(
           elementInstanceRecord.getProcessInstanceKey(), elementInstanceRecord.getElementId());
     }
 
+    final boolean hasProcessIncident =
+        incidentState.getProcessInstanceIncidentKey(elementInstance.getKey()) != MISSING_INCIDENT;
+    final boolean hasJobIncident =
+        incidentState.getJobIncidentKey(elementInstance.getJobKey()) != MISSING_INCIDENT;
```

Comment:
🔧 This lookup is only necessary if `elementInstance.getJobKey() > -1`

Reply:
@korthout I thought about the same thing but it looked a bit ugly in an if statement or when you do a separate if checks. We can refactor it something like:

```java
final boolean hasIncident =
        incidentState.getProcessInstanceIncidentKey(elementInstance.getKey()) != MISSING_INCIDENT ||
        incidentState.getJobIncidentKey(elementInstance.getJobKey()) != MISSING_INCIDENT;
```
WDYT?

Reply:
💭 Performance > Looks

```java
final boolean hasIncident =
        incidentState.getProcessInstanceIncidentKey(elementInstance.getKey()) != MISSING_INCIDENT ||
        (elementInstance.getJobKey() > -1L && incidentState.getJobIncidentKey(elementInstance.getJobKey()) != MISSING_INCIDENT;)
```

Reply:
Ahh, that is even better! 🚀

Reply:
Updated!